subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c

```diff
@@ -295,6 +295,11 @@ static int create_prepare_cb(struct lll_prepare_param *p)
 		radio_switch_complete_and_disable();
 	}
 
+	/* RSSI enable must be called after radio_switch_XXX function because it clears
+	 * RADIO->SHORTS register, thus disables all other shortcuts.
+	 */
+	radio_rssi_measure();
+
```

Comment:
Is this always true? If so, would it make sense to move `radio_rssi_measure` to the `radio_switch_*` functions?

Reply:
Well I would not put it into `radio_switch_*` to do not violate single responsibility principle. Besides that it isn't always enabled e.g.
https://github.com/zephyrproject-rtos/zephyr/blob/4f2762d7dabe076252cb3cfa31c1f9a1ab1322d3/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c#L1116-L1119

Reply:
That makes sense, but it could also be argued that if you do X, then you shall always do Y, then perhaps that should be a single function Z, e.g.

```
static void foo(void)
{
    /* Do foo */
}

static void barvoid)
{
    /* Do bar */
}

static void foobar(void)
{
    /* foo shall always be called before bar, and bar shall always be called after foo */
    foo();
    bar();
}
```

Reply:
Yes, that is of course true. I read your suggestion a bit too strictly as move `radio_rssi_measure` into `radio_switch_*` but without change of name of latter one.

I think that suggested change should be considered as an enhancement to whole LLL and more than `radio_rssi_measure` could be moved into some kind of `radio_configure_rx` or `radio_configure_tx`.